src/main/java/picard/sam/BamErrorMetric/CollectBamErrorMetrics.java

```diff
@@ -0,0 +1,470 @@
+    static private boolean advanceIteratorAndCheckLocus(final PeekableIterator<VariantContext> vcfIterator, final Locus locus, final SAMSequenceDictionary sequenceDictionary) {
+        while (vcfIterator.hasNext() && (vcfIterator.peek().isFiltered() ||
+                CompareVariantContextToLocus(sequenceDictionary, vcfIterator.peek(), locus) < 0)) {
+            vcfIterator.next();
+        }
+        return vcfIterator.hasNext() && CompareVariantContextToLocus(sequenceDictionary, vcfIterator.peek(), locus) == 0;
+    }
+
+    /**
+     * Iterate over the different records in the locus and add bases to aggregators
+     */
+    private void addLocusBases(final Collection<BaseErrorAggregation> aggregatorList, final SAMLocusAndReference info) {
+        info.getRecordAndOffsets()
+                .forEach(rao -> aggregatorList
+                        .forEach(l -> l.addBase(rao, info)));
+    }
+
+    /**
+     * Parses the list of "directives" and creates a collection of {@link BaseErrorAggregation}s based on them.
+     *
+     * @return Collection of {@link BaseErrorAggregation}
+     * @throws IllegalArgumentException if directives cannot be parsed or objects cannot be created.
+     */
+
+    private Collection<BaseErrorAggregation> getAggregatorList() {
+        final List<BaseErrorAggregation> aggregatorList = new ArrayList<>();
+        Set<String> suffixes = new HashSet<>();
+
+        ReadBaseStratification.setLongHomopolymer(LONG_HOMOPOLYMER);
+        for (final String directive : ERROR_METRICS) {
+            final BaseErrorAggregation aggregator;
+            aggregator = parseDirective(directive);
+            aggregatorList.add(aggregator);
+            if (!suffixes.add(aggregator.getSuffix())) {
+                throw new IllegalArgumentException(String.format("Duplicated suffix (%s) found in aggregator %s.",
+                        aggregator.getSuffix(), aggregator.getClass().toString()));
+            }
+        }
+        return aggregatorList;
+    }
+
+    /**
+     * Reads Interprets intervals from the input INTERVALS, if there's a file with that name, it opens the file, otherwise it tries to parse it, checks that their dictionaries all agree with the input SequenceDictionary
+     * and returns the intersection of all the lists.
+     *
+     * @param sequenceDictionary
+     * @return the intersection of the interval lists pointed to by the input parameter INTERVALS
+     */
+    private IntervalList getIntervals(final SAMSequenceDictionary sequenceDictionary) {
+
+        IntervalList regionOfInterest = null;
+
+        for (final File intervalListFile : INTERVALS) {
+
+            final IntervalList temp;
+            if (intervalListFile.exists()) {
+                log.info("Reading IntervalList ", intervalListFile, ".");
+                temp = IntervalList.fromFile(intervalListFile);
+                sequenceDictionary.assertSameDictionary(temp.getHeader().getSequenceDictionary());
+            } else {
```

Comment:
if you flip the `if` then you can make this just `if (!exists()) { .. throw }` and put everything else at the top level in the `for`. That would avoid having to declare `temp` before it's initialized.